src/content/en/fundamentals/instant-and-offline/web-storage/cache-api.md

```diff
@@ -0,0 +1,248 @@
+      <td>Interprets the bytes of the body as an HTML form, encoded either as "multipart/form-data"
+      or "application/x-www-form-urlencoded". Returns a
+      <a href="https://developer.mozilla.org/en-US/docs/Web/API/FormData">FormData</a> object, or
+      throws a <code>TypeError</code> if the data cannot be parsed.</td>
+    </tr>
+    <tr>
+      <td><code>body</code></td>
+      <td>Returns a <a      href="https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream">
+      ReadableStream</a> for the body data.</td>
+    </tr>
+  </tbody>
+</table>
+
+For example
+
+    const response = new Response('Hello world');
+    response.arrayBuffer().then((buffer) => {
+      console.log(new Uint8Array(buffer));
+      // Uint8Array(11) [72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100]
+    });
+
+## Creating and opening a cache
+
+To open a cache, use the `caches.open(name)` method, passing the name of the cache as the single
+parameter. If the named cache does not exist it will be created. This method returns a `Promise`
+that will resolve with the `Cache` object.
+
+    caches.open('my-cache).then((cache) => {
+      // do something with cache...
+    });
+
+## Retrieving from a cache
+
+To find an item in a cache, you can use the `match` method.
+
+    cache.match(request).then((response) => console.log(request, response));
+
+If `request` is a string it will first be converted to a `Request` by calling `new
+Request(request)`. The function returns a `Promise` that resolves to a `Response` if a matching
+entry is found, or `undefined` otherwise.
+
+To determine if two
+`Requests` match, more than just the URL is used. Two requests will be considered different
+if they have different query strings, `Vary` headers and/or methods (`GET`, `POST`, `PUT`, etc.).
+
+You can ignore some or all of these things by passing an options object as a second parameter.
+
+    const options = {
+      ignoreSearch: true,
+      ignoreMethod: true,
+      ignoreVary: true
+    };
+
+    cache.match(request, options).then(...);
+
+If more than one cached request matches then the one that was created first will be returned.
+
+If you want to retrieve *all* matching responses, you can use `cache.matchAll`.
+
+    const options = {
+      ignoreSearch: true,
+      ignoreMethod: true,
+      ignoreVary: true
+    };
+
+    cache.matchAll(request, options).then((responses) => {
+      console.log(`There are ${responses.length} matching responses.`);
+    });
+
+As a shortcut you can search over all caches at once by using `caches.match()` instead of calling
+`cache.match()` for each cache.
+
+## Searching
+
+The Cache API does not provide a way to search for requests or responses except for matching entries
+against a `Response` object. However, you can implement your own search using the features that are
+available.
+
+### Filtering
+
+One way is to iterate over all entries and filter down to the ones that you want. Let's say that you
```

Comment:
Link your first sentences back up to the intro, something like:

One way to implement your own search is to iterate over all the entries...